Open Judge System/OJS.Services/Data/ProblemGroups/ProblemGroupsDataService.cs

```diff
@@ -23,10 +27,15 @@ public class ProblemGroupsDataService : IProblemGroupsDataService
             this.problemGroups.All().Where(pg => pg.ContestId == contestId);
 
         public IQueryable<Problem> GetProblemsById(int id) =>
-            this.GetByIdQuery(id)
-                .SelectMany(eg => eg.Problems)
-                .Where(p => !p.IsDeleted);
+            this.GetByIdQuery(id).SelectMany(eg => eg.Problems).Where(p => !p.IsDeleted);
+
+        public void Delete(ProblemGroup problemGroup)
+        {
+            this.problemGroups.Delete(problemGroup);
```

Comment:
Might actually be a bad idea to have these methods, since confusion may arise when the same/similiar methods exist in the business service and they actually function according to the business logic. A better idea might be to not have these methods and have the actuall delete calls from the business service methods be done on the repository instead.